### PR TITLE
Changes to ButinaSplitter

### LINKED ISSUE #2198: Changes to ButinaSplitter
cc @rbharath @proteneer 

This is a followup to https://github.com/deepchem/deepchem/issues/2112#issuecomment-700942341 and the discussion following it.  ButinaSplitter is currently not very useful, but could be made more useful with a few changes.

As discussed there I modified it to use the same code for building datasets as ScaffoldSplitter.  I then tested it with the following code.

```python
splitters = ['random', 'scaffold', 'butina']
metric = dc.metrics.Metric(dc.metrics.roc_auc_score)
for splitter in splitters:
    tasks, datasets, transformers = dc.molnet.load_tox21(featurizer='ECFP', split=splitter)
    train_dataset, valid_dataset, test_dataset = datasets
    model = dc.models.MultitaskClassifier(n_tasks=len(tasks), n_features=1024, layer_sizes=[1000])
    model.fit(train_dataset, nb_epoch=10)
    print('splitter:', splitter)
    print('training set score:', model.evaluate(train_dataset, [metric], transformers))
    print('test set score:', model.evaluate(test_dataset, [metric], transformers))
    print()
```

RandomSplitter gives a test set score of 0.78, and ScaffoldSplitter gives 0.68, which shows how random splitting doesn't produce a truly independent test set.  Scaffold splitting does much better.  ButinaSplitter gives a test set score of 0.78, exactly the same as RandomSplitter!

ButinaSplitter has an adjustable cutoff.  The above was with the default value of 0.18.  So I repeated the test with different values of the cutoff.  Here are the test set scores.

0.18: 0.78
0.3: 0.74
0.4: 0.73
0.5: 0.66
0.6: 0.61
0.7: 0.65
0.8: 0.61

At least on this dataset, 0.18 is much too small.  It works no better than random splitting.  It gets better as you increase the cutoff, and by 0.6 it's significantly better than ScaffoldSplitter.  Further increases make little difference.

I looked up the original paper on the clustering method (http://www.l4patterns.com/uploads/dbclus-paper.pdf), and they used a cutoff of 0.8.  How was the value 0.18 selected?  Would it be reasonable to increase the default to 0.6?  Is this value likely to be universal, or will it vary a lot between datasets?

### PR DESCRIPTION
**Pull request by peastman**
Fixes #2198.  I made the following changes.

1. It now performs the splitting using identical code to ScaffoldSplitter, just based on clusters instead of scaffolds.
2. I increased the default cutoff to 0.6.
3. The cutoff is now a constructor argument.  This allows `split()` to have an identical signature to other splitters.  It also will allow you to control the cutoff for molnet loader function with, for example, `splitter=dc.split.ButinaSplitter(0.7)`.
4. I updated the splitters tutorial to discuss it as well.

### CODE REVIEW

**Review by rbharath — APPROVED** [review]
Looks good! Feel free to merge when ready :)
